### ISSUE TITLE
How to deal with 'AssertionError:10 mean matching returned malformed array'? 

### ISSUE THREAD

**Issue by sshah99** [AnotherSamWilson/miceforest#73]
I am trying to impute variables with multiple categories (few more than 500). I am getting the above mention error and could not find the solution in documentation. Any insight would helpful :).

**Comment by AnotherSamWilson** [comment]
You are probably running into this issue here:
https://github.com/AnotherSamWilson/miceforest/issues/49

How many rows are in your dataset? In practice, 500 categories tends to be too many to model using lightgbm. This will result in a # rows by 500 column array of predictions to run mean matching on. lightgbm grows a model for every single class, and then takes a softmax to get the final predictions - with 500 categories, many of those trees will not grow because there is no credible data to act on, in which case the probability will be 0.0, which results in singularity issues when calculating the logodds.

So, it is best to group these categories up. Using this many categories as a feature in a lightgbm model works out fine because of how lightgbm treats categorical features. Actually modeling 500 categories is a different story - much harder problem, not advised.
